src/ui/src/core_components/content/CoreAnnotatedText.vue

```diff
@@ -0,0 +1,339 @@
+			>
+				{{ content[0] }}
+				<span v-if="content[1]" class="annotation-subject">{{
+					getShortName(content[1])
+				}}</span>
+			</span>
+		</span>
+		<template v-if="fields.copyButtons.value === 'yes'">
+			<div class="controls">
+				<button
+					class="control-button"
+					@click="copyJSON(fields.text.value)"
+				>
+					Copy JSON
+				</button>
+				<button
+					class="control-button"
+					@click="copyText(fields.text.value)"
+				>
+					Copy raw
+				</button>
+			</div>
+		</template>
+	</div>
+</template>
+
+<script lang="ts">
+import { cssClasses, primaryTextColor } from "../renderer/sharedStyleFields";
+export default {
+	writer: {
+		name: "Annotated text",
+		description: "Shows text with annotations",
+		category: "Content",
+
+		fields: {
+			text: {
+				name: "KeyValue",
+				type: FieldType.Object,
+				desc: "Value array with text/annotations. Must be a JSON string or a state reference to an array.",
+				default: `["This ",["is", "Verb"]," some ",["annotated", "Adjective"],    ["text", "Noun"]," for those of ",["you", "Pronoun"]," who ",["like", "Verb"]," this sort of ",["thing", "Noun"],". ","And here's a ",["word", "", "#faf"]," with a fancy background but no label."]`,
+			},
+			referenceColor: {
+				name: "Reference",
+				desc: "The colour to be used as reference for chroma and luminance, and as the starting point for hue rotation.",
+				type: FieldType.Color,
+				default: "#5551FF",
+				category: FieldCategory.Style,
+			},
+			seed: {
+				name: "Seed value",
+				desc: "Choose a different value to reshuffle colours.",
+				type: FieldType.Number,
+				default: "1",
+				category: FieldCategory.Style,
+			},
+			rotateHue: {
+				name: "Rotate hue",
+				desc: "If active, rotates the hue depending on the content of the string. If turned off, the reference colour is always used.",
+				type: FieldType.Text,
+				options: {
+					yes: "yes",
+					no: "no",
+				},
+				default: "no",
+				category: FieldCategory.Style,
+			},
+			copyButtons: {
+				name: "Copy buttons",
+				desc: "If active, adds a control bar with both copy text and JSON buttons.",
+				type: FieldType.Text,
+				options: {
+					yes: "yes",
+					no: "no",
+				},
+				default: "no",
+				category: FieldCategory.Style,
+			},
+			primaryTextColor,
+			cssClasses,
+		},
+		previewField: "text",
+	},
+};
+</script>
+<script setup lang="ts">
+import { FieldCategory, FieldType } from "../writerTypes";
+import injectionKeys from "../injectionKeys";
+import { inject } from "vue";
+import chroma, { Color } from "chroma-js";
+
+const fields = inject(injectionKeys.evaluatedFields);
+
+const COLOR_STEPS = [
+	{ h: -78, s: -34, l: 16 },
+	{ h: -61, s: -37, l: 16 },
+	{ h: -2, s: 0, l: 24 },
+	{ h: -12, s: 0, l: 29 },
+	{ h: 28, s: -20, l: 24 },
+	{ h: -61, s: -95, l: 25 },
+	{ h: -173, s: 0, l: 16 },
+	{ h: -228, s: 0, l: 22 },
+	{ h: 69, s: 0, l: 25 },
+	{ h: 70, s: 0, l: 29 },
+];
```

Comment:
Think about how to extend to support more colors, maybe consider discussing this with Miffy. Right now, I'm using the Tag colors as specified in the design system.

Reply:
@ramedina86 I've asked Miffy a while ago on the additional colors and she pointed to the other color groups. The problem is that they don't match 😅 I was waiting for Pete being in the office to ambush him but he keeps avoiding me so far.

It seems like it's either we're going with 10 colors or an LLM is out best bet 🤣